### PR TITLE
C307 autogen subdomains

### PR DESCRIPTION
**Pull request by ssickles**
This PR changes how this module is designed to work. In the new design, a subdomain is configured in a stack as a block. Then, if the module for the block is this module, it will create a new autogen_subdomain and delegate it for each environment.

In order to identify the autogen_subdomain to create and delegate we first pull in the subdomain block and get its id. We already have the env from the ns_workspace. The last piece is to make sure the ns_autogen_subdomain and ns_autogen_subdomain_delegation resources are passed these values so they can create and update accordingly.

### CODE REVIEW

File: subdomain.tf [hunk 1]
```diff
@@ -1,11 +1,20 @@
-resource "ns_autogen_subdomain" "subdomain" {}
+data "ns_subdomain" "subdomain" {
```
Comment (BSick7) [inline subdomain.tf:1]: As discussed earlier, could collapse this to `data "ns_block" "this"`. Did you want to do that in this PR or another PR?

**Review by BSick7 — CHANGES_REQUESTED** [review]
Looks good. I usually make a habit of running `terraform fmt` before commit.

File: subdomain.tf (hunk 1 above)
Comment (ssickles) [reply subdomain.tf:1]: Yes, I'm going to do this in the next PR.

**Comment by ssickles** [comment]
> Looks good. I usually make a habit of running `terraform fmt` before commit.

I went ahead and ran `terraform fmt` and that fixed some indenting.
